.travis.yml

```diff
@@ -169,13 +169,39 @@ matrix:
           find "(" -name "*.a" -or -name "*.ar" ")" -and -not -name "lib*" |
           tee BUILD/badlibs |
           sed -e "s/^/Bad library name found: /" && [ ! -s BUILD/badlibs ]
-        # Assert that all assebler files are named correctly
+        # Assert that all assembler files are named correctly
         # The strange command below asserts that there are exactly 0 libraries
         # that do end with .s
         - >
           find  -name "*.s" | tee BUILD/badasm |
           sed -e "s/^/Bad Assembler file name found: /" && [ ! -s BUILD/badasm ]
 
+    - <<: *docs-vm
+      name: "Frozen tools check"
+      env: NAME=frozen_tools_check
+      script:
+        # Reject any changes to tools that would require a re-release of the
+        # tools for the online compiler.
+        - >-
+            frozen_files=`\
+                git diff --name-only --diff-filter=d FETCH_HEAD..HEAD \
+                    | egrep \
+                      -e "^tools/build_api*" \
+                      -e "^tools/config*" \
+                      -e "^tools/export*" \
+                      -e "^tools/notifier*" \
+                      -e "^tools/paths*" \
+                      -e "^tools/resources*" \
+                      -e "^tools/targets*" \
+                      -e "^tools/toolchains*" \
+                      -e "^tools/utils*" \
+                      -e "^$"`
+            if [ -z "$frozen_files" ]; then
+                echo "Success!";
+            else
+                echo -e "Failure: Frozen files were modified\n$frozen_files";
```

Comment:
Do we have a documentation we could point to here if not including more detailed message? Once it errors, we should have a reference why we froze and what it means for the future (for other branches as well, this is for master, same will be applied to 5.15 branch?). Shall we consider a blog post, including message about new build tools?

There is at least one opened PR that will be affected.

Reply:
Tools changes to 5.15 can go ahead, but they may not have the desired effect. They'll affect the offline tooling only. Any new features added there which Mbed OS 5.15.x would depend on, or any bug fixes made there, wouldn't become available in the Online Compiler. The Mbed Online Compiler uses only the latest tools, from `mbed-os` `master`.

Yes, we should do a blog post. I'll write one. Then we can link to it from the CI failure message. Let's hold off merging this PR until we have that blog post up.

Reply:
Blog post is live at https://os.mbed.com/blog/entry/Introducing-the-new-Mbed-Tools/